File: FluentFTP.VBExamples/GetWorkingDirectory.vb
Imports System
Imports System.Net
Imports System.Threading
Imports System.Threading.Tasks
Imports FluentFTP

Namespace Examples
	Friend Module GetWorkingDirectoryExample
		Sub GetWorkingDirectory()
			Using conn = New FtpClient("127.0.0.1", "ftptest", "ftptest")
				conn.Connect()
				Console.WriteLine("The working directory is: " & conn.GetWorkingDirectory())
			End Using
		End Sub

		Async Function GetWorkingDirectoryAsync() As Task
			Dim token = New CancellationToken()

			Using conn = New AsyncFtpClient("127.0.0.1", "ftptest", "ftptest")
				Await conn.Connect(token)
				Console.WriteLine("The working directory is: " & Await conn.GetWorkingDirectory(token))
			End Using
		End Function
	End Module
End Namespace

File: FluentFTP.CSharpExamples/GetWorkingDirectory.cs
using System;
using System.Net;
using System.Threading;
using System.Threading.Tasks;
using FluentFTP;

namespace Examples {
	internal static class GetWorkingDirectoryExample {

		public static void GetWorkingDirectory() {
			using (var conn = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				conn.Connect();

				Console.WriteLine("The working directory is: " + conn.GetWorkingDirectory());
			}
		}

		public static async Task GetWorkingDirectoryAsync() {
			var token = new CancellationToken();
			using (var conn = new AsyncFtpClient("127.0.0.1", "ftptest", "ftptest")) {
				await conn.Connect(token);

				Console.WriteLine("The working directory is: " + await conn.GetWorkingDirectory(token));
			}
		}

	}
}